api-docs.js

```diff
@@ -0,0 +1,160 @@
+
+const apiDoc = {
+  openapi: '3.0.0',
+  servers: [
+    { url: "http://localhost:8000" }
+  ],
+  info: {
+    title: 'Headless Hathor Wallet API',
+    description: 'This wallet is fully controlled through an HTTP API.',
+    version: '1.0.0',
+  },
+  produces: [ "application/json" ],
+  paths: {
+    '/status': {
+      get: {
+        summary: 'Return the wallet status',
+        responses: {
+          200: {
+            description: 'A JSON with general wallet info',
+            content: {
+              'application/json': {
+                examples: {
+                  closed: {
+                    summary: 'Wallet is closed',
+                    value: {"statusCode":0,"statusMessage":"Closed","network":"mainnet","serverUrl":"https://node2.mainnet.hathor.network/v1a/","serverInfo":null}
+                  },
+                  connecting: {
+                    summary: 'Wallet is connecting',
+                    value: {"statusCode":1,"statusMessage":"Connecting","network":"mainnet","serverUrl":"https://node2.mainnet.hathor.network/v1a/","serverInfo":null}
+                  },
+                  syncing: {
+                    summary: 'Wallet is syncing',
+                    value: {"statusCode":2,"statusMessage":"Syncing","network":"mainnet","serverUrl":"https://node2.mainnet.hathor.network/v1a/","serverInfo":{"version":"0.29.0","network":"mainnet","min_weight":14,"min_tx_weight":14,"min_tx_weight_coefficient":1.6,"min_tx_weight_k":100,"token_deposit_percentage":0.01}}
+                  },
+                  ready: {
+                    summary: 'Wallet is ready',
+                    value: {"statusCode":3,"statusMessage":"Ready","network":"mainnet","serverUrl":"https://node2.mainnet.hathor.network/v1a/","serverInfo":{"version":"0.29.0","network":"mainnet","min_weight":14,"min_tx_weight":14,"min_tx_weight_coefficient":1.6,"min_tx_weight_k":100,"token_deposit_percentage":0.01}}
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+    '/0/balance': {
```

Comment:
Why `/0/`? Is it the token UID?

Reply:
I though that it may have multiple wallets. Maybe using a different derivation in the HD path, or instantiating more wallets. It would be nice if the wallets could share one connection.

I also used it to put a middleware that checks if the wallet is ready. If it is not, it returns an error.

Reply:
Humm understand. I think it's ok